src/transformers/models/modernbert/modular_modernbert.py

```diff
@@ -0,0 +1,1515 @@
+        attention, we re-unpad our outputs for the other layers. The pad/unpad operations add overhead, but not
+        sending pad tokens through ffs saves compute.
+
+        Args:
+            hidden_states: (total_nnz, dim)
+            cu_seqlens: (batch + 1,)
+            max_seqlen: int
+            indices: (total_nnz,)
+            attn_mask: (batch, max_seqlen)
+
+        Returns:
+            attention: (total_nnz, dim)
+        """
+        qkv = self.Wqkv(hidden_states)
+
+        attn_kwargs = {
+            "output_attentions": output_attentions,
+            "dim": self.all_head_size,
+        }
+        if self.config._attn_implementation == "flash_attention_2":
+            bs = hidden_states.shape[0]
+            qkv = qkv.view(-1, 3, self.num_heads, self.head_dim)
+
+            attn_kwargs.update(
+                {
+                    "local_attention": self.local_attention,
+                    "cu_seqlens": cu_seqlens,
+                    "max_seqlen": max_seqlen,
+                    "bs": bs,
+                }
+            )
+        else:
+            bs, seqlen = hidden_states.shape[:2]
+            qkv = qkv.view(bs, seqlen, 3, self.num_heads, self.head_dim)
+
+            attn_kwargs.update(
+                {
+                    "position_ids": position_ids,
+                    "attention_mask": attention_mask,
+                    "bs": bs,
+                    "seqlen": seqlen,
+                }
+            )
+
+        if output_attentions:
+            if self.config._attn_implementation == "sdpa":
+                logger.warning_once(
+                    "Outputting attentions is only supported with the 'eager' attention implementation, "
+                    'not with "sdpa". Falling back to `attn_implementation="eager"`.'
+                )
+                self.config._attn_implementation = "eager"
+            elif self.config._attn_implementation != "eager":
+                logger.warning_once(
+                    "Outputting attentions is only supported with the eager attention implementation, "
+                    f'not with {self.config._attn_implementation}. Consider setting `attn_implementation="eager"`.'
+                    " Setting `output_attentions=False`."
+                )
+
+        attn_outputs = MODERNBERT_ATTENTION_FUNCTION[self.config._attn_implementation](
+            self,
+            qkv=qkv,
+            rotary_emb=self.rotary_emb,
+            **attn_kwargs,
+        )
+        hidden_states = attn_outputs[0]
+
+        return (self.out_drop(self.Wo(hidden_states)),) + attn_outputs[1:]  # add attentions if outputted
+
+
+class ModernBertEncoderLayer(nn.Module):
+    def __init__(self, config: ModernBertConfig, layer_id: Optional[int] = None):
+        super().__init__()
+        self.config = config
+        if config.skip_first_prenorm and config.embedding_norm and layer_id == 0:
```

Comment:
are there checkpoints released with `skip_first_prenorm` ? We need to remove all code paths that don't correspond to released checkpoints!

Reply:
Actually, all release checkpoint will have `skip_first_prenorm` and `embedding_norm` as True, so I'll keep this one around.